### PR TITLE
#47 add application version detail view

### PR DESCRIPTION
**Pull request by elauria**

### CODE REVIEW

File: client/lib/application/src/version-detail/placeholder.hbs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<div class="applicationDetail u-placeholder">
+    <h1 class="u-placeholder-text">{{ver.application_id}} {{ver.id}}</h1>
```
Comment (prayerslayer) [inline client/lib/application/src/version-detail/placeholder.hbs]: Those things do not exist when the placeholder is shown, right?

File: client/lib/application/src/version-detail/version-detail.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import BaseView from 'common/src/base-view';
+import Template from './version-detail.hbs';
+import Flux from 'application/src/flux';
+import FetchResult from 'common/src/fetch-result';
+import ErrorTpl from 'common/src/error.hbs';
+import Placeholder from './placeholder.hbs';
+import Markdown from 'common/src/markdown';
+
+import 'common/asset/scss/application/application-detail.scss';
```
Comment (prayerslayer) [inline client/lib/application/src/version-detail/version-detail.js]: Should be `version-detail` I suppose?

File: client/lib/application/src/version-detail/version-detail.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+class VersionDetail extends BaseView {
+    constructor( props ) {
+        this.stores = {
+            application: Flux.getStore('application'),
```
Comment (prayerslayer) [inline client/lib/application/src/version-detail/version-detail.js]: Trailing comma ;)

File: client/lib/application/src/version-detail/version-detail.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        };
+        this.className = 'applicationDetail';
```
Comment (prayerslayer) [inline client/lib/application/src/version-detail/version-detail.js]: This should also be `versionDetail`.

File: client/lib/application/src/version-detail/placeholder.hbs [hunk 5]
```diff
@@ -0,0 +1,21 @@
+<div class="applicationDetail u-placeholder">
```
Comment (prayerslayer) [inline client/lib/application/src/version-detail/placeholder.hbs]: Wrong class :)

File: client/lib/application/src/version-detail/version-detail.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        super(props);
+    }
+
+    update() {
+        let {applicationId} = this.props,
+            {versionId} = this.props;
+        this.data = {
+            ver: this.stores.application.getApplicationVersion( applicationId, versionId )
+        };
+        console.log('ver', this.data.ver);
```
Comment (prayerslayer) [inline client/lib/application/src/version-detail/version-detail.js]: We shouldn’t output logs after we got things to work.

File: client/lib/application/src/version-detail/version-detail.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        };
+        this.className = 'applicationDetail';
+        super(props);
+    }
+
+    update() {
+        let {applicationId} = this.props,
+            {versionId} = this.props;
+        this.data = {
+            ver: this.stores.application.getApplicationVersion( applicationId, versionId )
```
Comment (prayerslayer) [inline client/lib/application/src/version-detail/version-detail.js]: This is a bit nitpicky, but would you mind changing this variable to `version`? It just reads better in the template.

File: client/lib/common/src/data/application/application-store.js [hunk 8]
```diff
@@ -62,6 +68,20 @@ class ApplicationStore extends Store {
     }
 
     /**
+     * Replaces application with `id` and version `ver` with a Pending state.
+     *
+     * @param  {String} id
+     * @param  {String} ver
+     */
+    beginFetchApplicationVersion( id, ver ) {
+        let updatedApp =  _m.assoc( _m.get(this.state.versions, id) || _m.hashMap(), ver, new Pending() )
```
Comment (prayerslayer) [inline client/lib/common/src/data/application/application-store.js]: Missing semicolon.

File: client/lib/common/src/data/application/application-store.js [hunk 9]
```diff
@@ -73,6 +93,17 @@ class ApplicationStore extends Store {
     }
 
     /**
+     * Replaces application version with Failed state.
+     *
+     * @param  {error} err The error passed from the flux action.
+     */
+    failFetchApplicationVersion(err) {
+        this.setState({
+            versions: _m.assoc( this.state.versions, err.ver, new Failed( err ) )
```
Comment (prayerslayer) [inline client/lib/common/src/data/application/application-store.js]: Here you replace the application `ver` with a Failed result, which is probably not what you wanted.

File: client/lib/application/src/version-detail/version-detail.js (hunk 2 above)
Comment (elauria) [inline client/lib/application/src/version-detail/version-detail.js]: as currently has no differences with application-detail, i decided to not replicate code.

File: client/lib/common/src/data/application/application-store.js (hunk 9 above)
Comment (elauria) [inline client/lib/common/src/data/application/application-store.js]: not sure what should be done here.. failFetchApplication is replacing the application id with a Failed result

**Comment by prayerslayer** [comment]
Looks good :+1:

**Pull request merged**